test/buffer-source.js

```diff
@@ -0,0 +1,125 @@
+"use strict";
+
+const assert = require("assert");
+
+const conversions = require("..");
+
+function commonNotOk(sut) {
+    it("should throw a TypeError for `undefined`", () => {
+        assert.throws(() => sut(undefined), TypeError);
+    });
+
+    it("should throw a TypeError for `null`", () => {
+        assert.throws(() => sut(null), TypeError);
+    });
+
+    it("should throw a TypeError for `true`", () => {
+        assert.throws(() => sut(true), TypeError);
+    });
+
+    it("should throw a TypeError for `false`", () => {
+        assert.throws(() => sut(false), TypeError);
+    });
+
+    it("should throw a TypeError for `Infinity`", () => {
+        assert.throws(() => sut(Infinity), TypeError);
+    });
+
+    it("should throw a TypeError for `NaN`", () => {
+        assert.throws(() => sut(NaN), TypeError);
+    });
+
+    it("should throw a TypeError for `0`", () => {
+        assert.throws(() => sut(0), TypeError);
+    });
+
+    it("should throw a TypeError for `''`", () => {
+        assert.throws(() => sut(""), TypeError);
+    });
+
+    it("should throw a TypeError for `Symbol.iterator`", () => {
+        assert.throws(() => sut(Symbol.iterator), TypeError);
+    });
+
+    it("should throw a TypeError for `{}`", () => {
+        assert.throws(() => sut({}), TypeError);
+    });
+
+    it("should throw a TypeError for `() => {}`", () => {
+        assert.throws(() => sut(() => {}), TypeError);
+    });
+}
+
+function testOk(name, sut, create) {
+    it("should return `" + name + "` object for `" + name + "` object", () => {
+        const obj = create();
+        assert.strictEqual(sut(obj), obj);
+    });
+}
+
+function testNotOk(name, sut, create) {
+    it("should throw a TypeError for `" + name + "` object", () => {
+        assert.throws(() => sut(create()), TypeError);
+    });
+}
+
+const objCreators = [
+    [ DataView, () => new DataView(new ArrayBuffer(0)) ]
```

Comment:
How about using `{ constructor, creator }` object literals instead of arrays? I don't think the brevity savings here and on line 82 are worth the loss of clarity on lines 91/92 and friends.